トップページの仕様変更

Monma9955 opened the pull request:
## What
- ビュー変更：
1. ルートパスをcategories#indexに変更
1. トップページのサイドバーの中身をルートカテゴリ名の一覧に変更
1. categories#showのビューを作成
1. categories#show_articlesのビューを作成

- コントローラー追加：categoriesコントローラー追加
- スクショ：
1. トップページ https://gyazo.com/70d214f77d74c2164fe8e9d74f15e963
1. カテゴリ選択時 https://gyazo.com/7444c38861e80ed0fa8c0d308b407280

- その他：scssファイル管理用にpartialsディレクトリ作成。コントローラーごとにscssファイルを格納。

## Why
- 問題：カテゴリの一覧から設問を探す場合、root→parents→children階層のカテゴリをそれぞれ表示させないといけないため、１ページで実装しようとするとユーザビリティが悪くなってしまう。
- 原因：メイン画面だけだと表示領域が狭いため。
- 解決方法：ルートカテゴリの選択をサイドバーでできるようにした。副次的な効果としてログインメニューを廃したことでユーザーは設問を探すことに集中できるようになった。

Monma9955 committed:
categories_controller作成

トップページの仕様変更。
トップページをcategories#indexのビューにするためにrails gコマンドでカテゴリコントローラーを作成しました。

Monma9955 committed:
rootパス変更 routes.rb

トップページをcategoriesコントローラーのindexビューに変更しました。
ビューはこれから作成。

Monma9955 committed:
ルートカテゴリ取得 categories#index

カテゴリコントローラーのindexアクションでルートカテゴリの一覧が使用できるようにしました。

Monma9955 committed:
トップページ部分テンプレート化

rootページをcategories#indexに変更。
ビューはarticles#indexを踏襲しつつ、ヘッダーとサイドバーは部分テンプレートに切り出しました。
元々のサイドバーはログインメニューとして使用し、トップページのサイドバーはトップページ用に変更予定(現在は全く同じ内容)。

Monma9955 committed:
各コントローラーのscssファイル構成整理

すべてのscssファイルをapplication.scssで読み込ませず、各コントローラーのscssファイルは一旦partialsディレクトリで管理するようにしました。
それに伴い、トップページのヘッダーとサイドバーのscssは別々に切り出しました。

Monma9955 committed:
user.scss移動

user.scssはユーザーコントローラーに関連するビューのscssファイルなので、partials配下に移動しました。

Monma9955 committed:
カテゴリサイドバーhaml作成

カテゴリ一覧が表示されるサイドバーの部分テンプレートを作成。
DBからルートカテゴリを取得して一覧表示できるようにしました。

Monma9955 committed:
サイドバーのscssファイルを統合

サイドバーのscssファイルを２つに分けていましたが、共通するプロパティをmixinに定義し、サイドバーのscssファイルは１つに統合しました。
上記に伴いログインメニューのサイドバーのid名もカテゴリのサイドバーと明示的に区別するように変更。

Monma9955 committed:
ignoreファイル追加

ローカルで使用したいファイルを格納しておけるignore_filesディレクトリを作成。
コミットされないようgitignoreファイルで指定しました。

Monma9955 committed:
トップページコンテンツ表示画面削除

トップページのメインコンテンツ表示画面を削除。
あてていたCSSは一旦コメントアウトしました。
サイドバーとメイン画面間のボーダーは廃止し、サイドバーにグラデーションの色をつけることにしました。

Monma9955 committed:
categories#show作成

categoriesコントローラーのshowアクションを作成。
ルーティングを組んでshow.html.hamlが表示されるようにしました。
show.html.hamlの内容は以前のindex.html.hamlと同一のためコピー＆ペーストで実装。

Monma9955 committed:
css切り出し　top.scss→show.scss

top.scss内でshow.html.hamlに移したクラスのcssはshow.scssファイル(を新しく作成して)に移動しました。

Monma9955 committed:
不要なモデルロジック削除 category.rb

categoryモデルのロジックが不要だったため削除。
self.root：元々rootsというメソッドがあり、役割が同じため
self.parent_categories：使用していないため

Monma9955 committed:
sider指摘箇所修正 routes.rb

rootにcategories#indexを定義しているため、resourcesでindexを定義しなくても動作しました。
routes.rbからindexの表記を削除。

Monma9955 committed:
sider指摘箇所修正 ビューのロジックをコントローラーに移動

siderで指摘のあったshow.html.hamlのロジックをコントローラーに移動させました。

Monma9955 committed:
seed 長過ぎるカテゴリ名修正 category.rb

「その他のWeb・IT・デザインスキル」のカテゴリ名が長すぎてトップページで文字が折り返されてしまったためカテゴリ名を省略しました。

Monma9955 committed:
サイドバーのheight変更

サイドバーのheightをulタグではなくmixinの最初に適用。
『サイドバーの高さ=100vhからヘッダーの高さを抜いた高さ』になるようにしました。

Monma9955 committed:
非同期のarticles表示 仮置のhtml、css作成

選択した親カテゴリに紐づく小カテゴリの問題が非同期で表示される機能の実装。
jQueryで表示させるhtmlをまずはhamlファイルに記述しました。

Monma9955 committed:
haml・scss変更 pageの要素削除 show.html.haml

設問一覧の表示で問題文(page)にタイトルのカラムがないのに問題文ごとのタイトル表示をするような表示になっていたため、一覧には問題のタイトルのみが出るように変更しました。

ghost commented:
Sider detected **1** warning on analyzing the commit 770a9db.

- ⚠️ [RuboCop](https://sider.review/gh/repos/263128061/pulls/13/logs/3fadf319-ce33-4bb9-b4cb-ac5cda8decce)

We recommend fixing them as possible by updating the dependencies, configuring the analysis tool, [configuring `sider.yml`](https://sider.review/gh/repos/263128061/pulls/13#config), [turning off unused tools](https://sider.review/gh/repos/263128061/settings), and so on.

If you have problems or questions still, feel free to ask us via chat. 💬

---

You can [turn off this notification](https://sider.review/gh/repos/263128061/settings#notifications) if you don't want to receive it from now on.

Monma9955 committed:
アコーディオンメニュー実装 categories.js

jQueryでアコーディオンメニューを実装。
.clickで子カテゴリのクラスを取得し、.nextにあたる問題一覧の表示非表示を切り替えられるようにしました。

Monma9955 committed:
console.log削除 articles.js

articles.jsに消し忘れたconsole.logが会ったので削除しました。

Monma9955 committed:
categories#show_articles作成

categories#showのメイン画面でparentカテゴリをクリックすると下に出てくるメニューをshow_articlesのビューで作成しました。

Monma9955 committed:
show_articlesにscss適用 link-color作成

show_scssのselected-parentから下のscssをshow_articlesのビュー用にしました。
画面遷移のリンクはリンク用の青色を作成して適用しました。

Monma9955 committed:
インスタンス変数作成 categories_controller.rb

categories#show_articlesに渡すインスタンス変数を作成しました。

Monma9955 committed:
仮置haml変更 部分テンプレート化 show_articles.html.haml

コントローラーにデータのインスタンス変数が定義されるまで仮置だったhamlを正式な仕様に変更。
各articleは読み込み速度向上のため部分テンプレート化しました。

Monma9955 committed:
UX向上 show.html.haml

サイドバーのrootカテゴリーを選択した際に表示される最初のparentカテゴリーのchildrenを最初に選択できるようにしました。

Monma9955 committed:
トップページ ビュー作成

トップページは新着の問題10件を表示させるよう表示を変更しました。

Monma9955 committed:
articleリンク化(仮) articles.html.haml

articles.html.hamlでarticleをリンクにしました。
